www/lib/lib.php

```diff
@@ -767,20 +767,35 @@ public function getInfo($id){
 			$this->ptzControl = new cameraPtz($this);
 		}
 	}
-	public function checkConnection(){ 
+	public function checkConnection() {
+		 //FIXME: Currently only supports RTSP. MJPEG is currently untested
 		ini_set('default_socket_timeout', 1);
-		#needs server to check for RTSP //// $paths['rtsp']  = 'http://'.((empty($this->info['rtsp_username'])) ? '' : $this->info['rtsp_username'].':'.$this->info['rtsp_password'].'@').$this->info['ipAddr'].':'.$this->info['port'].$this->info['rtsp'];
-		$paths['mjpeg'] = 'http://'.((empty($this->info['rtsp_username'])) ? '' : $this->info['rtsp_username'].':'.$this->info['rtsp_password'].'@').((empty($this->info['ipAddrMjpeg'])) ? $this->info['ipAddr'] : $this->info['ipAddrMjpeg']).':'.$this->info['portMjpeg'].$this->info['mjpeg_path'];
-		$paths['http'] = 'http://'.((empty($this->info['rtsp_username'])) ? '' : $this->info['rtsp_username'].':'.$this->info['rtsp_password'].'@').((empty($this->info['ipAddrMjpeg'])) ? $this->info['ipAddr'] : $this->info['ipAddrMjpeg']);
-		foreach($paths as $type => $path){
-			$headers = @get_headers($path);
-			$contents = @file_get_contents($path);
-			if (!$headers) { $this->info['connection_status'][$type] = 'F'; continue; }
-			preg_match("/([0-9]{3})/", $headers[0], $response_code);
-			$this->info['connection_status'][$type] = ($response_code[0]=='200') ? 'OK' : $response_code[0];
+
+		$path = "";
+
+		switch($this->info['protocol'])  {
+			case 'IP-RTSP':
+				$path = '-rtsp_transport tcp "rtsp://'.((empty($this->info['rtsp_username'])) ? '' : $this->info['rtsp_username'].':'.$this->info['rtsp_password'].'@').$this->info['ipAddr'].':'.$this->info['port'].$this->info['rtsp'].'"';
```

Comment:
Hand-coded command line argument quoting is unsafe.
`escapeshellarg($url)` is needed here.

Below suggestion block is not for immediate application, but to illustrate the point:
```suggestion
				$url = 'rtsp://'.((empty($this->info['rtsp_username'])) ? '' : $this->info['rtsp_username'].':'.$this->info['rtsp_password'].'@').$this->info['ipAddr'].':'.$this->info['port'].$this->info['rtsp'];
				$rtsp_options = '-rtsp_transport tcp'; // TODO according to device config, see https://github.com/bluecherrydvr/bluecherry-apps/blob/master/lib/lavf_device.cpp#L90
				$cmdline = "/usr/lib/bluecherry/ffprobe -stimeout 5000000 -hide_banner -show_format -show_streams -print_format json ' . $rtsp_options . ' ' . escapeshellarg($url);
```

Hand-coded URL composing seems to work as intended here. (Would preferred to see some "URL builder" API usage here, but couldn't actually find a good one in PHP in a brief search.)